Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/States/StateManager.cs

```diff
@@ -0,0 +1,422 @@
+        public InteractionState GetState(string stateName)
+        {
+            InteractionState interactionState = states.Find((state) => state.Name == stateName);
+
+            return interactionState;
+        }
+
+        /// <summary>
+        /// Gets a Core Interaction State.
+        /// </summary>
+        /// <param name="coreState">The CoreInteractionState to retrieve</param>
+        /// <returns>The Core Interaction State contained in the Tracked States scriptable object.</returns>
+        public InteractionState GetState(CoreInteractionState coreState)
+        {
+            return GetState(coreState.ToString());
+        }
+
+        /// <summary>
+        /// Gets and sets state given the state name and state value.
+        /// </summary>
+        /// <param name="stateName">The name of the state to set</param>
+        /// <param name="value">The new state value</param>
+        /// <returns>The state that was set</returns>
+        public InteractionState SetState(string stateName, int value)
+        {
+            InteractionState state = GetState(stateName);
+
+            if (state != null)
+            {
+                if (value > 0)
+                {
+                    SetStateOn(stateName);
+                }
+                else
+                {
+                    SetStateOff(stateName);
+                }
+            }
+            else
+            {
+                Debug.LogError($"The {stateName} state is not being tracked, add this state using AddState(state) to set it");
+            }
+
+            return state;
+        }
+
+        /// <summary>
+        /// Gets and sets a Core Interaction State given the value.
+        /// </summary>
+        /// <param name="coreState">The Core Interaction State to set</param>
+        /// <param name="value">The Core Interaction State's new value</param>
+        /// <returns>The Core Interaction State that was set</returns>
+        public InteractionState SetState(CoreInteractionState coreState, int value)
+        {
+            return SetState(coreState.ToString(), value);
+        }
+
+        /// <summary>
+        /// Gets and sets a state to On and invokes the OnStateActivated event. Setting a 
+        /// state on changes the state value to 1.
+        /// </summary>
+        /// <param name="stateName">The name of the state to set to on</param>
+        /// <returns>The state that was set to on</returns>
+        public InteractionState SetStateOn(string stateName)
+        {
+            InteractionState state = GetState(stateName);
+
+            if (state != null)
+            {
+                if (state.Value != 1)
+                {
+                    state.Value = 1;
+                    state.Active = 1;
+
+                    OnStateActivated.Invoke(state);
+
+                    // Only add the state to activeStates if it is not present 
+                    if (!activeStates.Contains(state))
```

Comment:
Consider using hashset instead of list here for better perf (not critical, since this probably won't run every frame and I assume the list will be short). HashSet will use O(1) for contains test where List will be O(N).